src/pint/models/stand_alone_psr_binaries/BT_piecewise.py

```diff
@@ -0,0 +1,441 @@
+import astropy.constants as c
+import astropy.units as u
+import numpy as np
+import pint.toa
+from pint import GMsun, Tsun, ls
+from pint.models.stand_alone_psr_binaries.BT_model import BTmodel
+from .binary_generic import PSR_BINARY
+
+
+class BTpiecewise(BTmodel):
+    def __init__(self, axis_store_initial=None, t=None, input_params=None):
+        self.binary_name = "BT_piecewise"
+        super(BTpiecewise, self).__init__()
+        self.axis_store_initial = []
+        self.extended_group_range = []
+        self.d_binarydelay_d_par_funcs = [self.d_BTdelay_d_par]
+        if t is not None:
+            self._t = t
```

Comment:
It's probably a good idea to set `selt._t` to None if `t` is None, rather than leaving it uninitialized.

Reply:
Changed as suggested